boot/bootutil/src/image_validate.c

```diff
@@ -88,6 +88,53 @@ bootutil_img_hash(struct image_header *hdr, const struct flash_area *fap,
 }
 
 /*
+ * Currently, we only support being able to verify one type of
+ * signature, because there is a single verification function that we
+ * call.  List the type of TLV we are expecting.  If we aren't
+ * configured for any signature, don't define this macro.
+ */
+#if defined(MCUBOOT_SIGN_RSA)
+#    define EXPECTED_SIG_TLV IMAGE_TLV_RSA2048
+#    define EXPECTED_SIG_LEN(x) ((x) == 256) /* 2048 bits */
+#    if defined(MCUBOOT_SIGN_EC) || defined(MCUBOOT_SIGN_EC256)
```

Comment:
This is a bit hard to read and might get harder if more signature types are supported.

What about:

```
#if (defined(MCUBOOT_SIGN_RSA) + defined(MCUBOOT_SIGN_EC) + defined(MCUBOOT_SIGN_EC256)) > 1
#error Multiple signature types not yet supported
#endif
```

Then you can just write:

```
#if defined(MCUBOOT_SIGN_RSA)
...
#elif defined(MCUBOOT_SIGN_EC)
... EXPECTED_SIG_TLV, EXPECTED_SIG_LEN for each case go here...
#elif defined(MCUBOOT_SIGN_EC256)
...
#endif
```